.erk/docs/agent/architecture/uvx-hooks.md

```diff
@@ -0,0 +1,121 @@
+---
+title: uvx Hook Invocation Pattern
+read_when:
+  - implementing plugin hooks with uvx
+  - understanding erk hook script architecture
+  - debugging hook version issues
+---
+
+# uvx Hook Invocation Pattern
+
+> **Note:** This documentation was produced in December 2025 based on Claude Code's plugin system at that time. The plugin system is actively evolving; verify against [official Claude Code documentation](https://docs.anthropic.com/en/docs/claude-code) for current behavior.
+
+ERK hooks use `uvx erk@{version} kit exec` for reproducible, version-pinned script execution.
+
+## The Pattern
+
+```json
+{
+  "type": "command",
+  "command": "uvx erk@1.2.3 kit exec erk session-id-injector-hook",
+  "timeout": 30
+}
+```
+
+## Why uvx?
+
+### Alternative 1: Bundle Scripts in Plugins
+
+```json
+{
+  "command": "python3 ${CLAUDE_PLUGIN_ROOT}/scripts/hook.py"
+}
+```
+
+**Problems:**
+
+- Scripts lose access to `erk_shared` (context, git/github ABCs)
+- No Click decorators (`@logged_hook`, `@project_scoped`)
+- Must vendor dependencies or require separate install
+- No type safety from erk's pyright config
+
+### Alternative 2: Require erk Installation
+
+```json
+{
+  "command": "erk kit exec erk session-id-injector-hook"
+}
+```
+
+**Problems:**
+
+- Requires erk to be installed globally
+- Version drift: user's erk version may differ from plugin expectation
+- No reproducibility guarantee
+
+### Alternative 3: uvx with Version Pinning (Chosen)
+
+```json
+{
+  "command": "uvx erk@1.2.3 kit exec erk session-id-injector-hook"
+}
+```
+
+**Benefits:**
+
+- **Version pinning**: Exact erk version, reproducible behavior
+- **No install required**: uvx resolves and caches automatically
+- **Full erk access**: Scripts use erk_shared, Click, type checking
+- **Fast**: ~50ms warm start, ~200ms cold start
+- **Ubiquitous**: uv is increasingly standard in Python ecosystem
+
+## How It Works
+
+1. Hook fires in Claude Code
+2. uvx downloads/caches erk@1.2.3 if not present
```

Comment:
Instead of framing this as a chosen alternative, present it as what we do up front and then explain the tradeoffs against the alternatives we did NOT chose after explain our architecture